src/api/java/com/minecolonies/api/colony/interactionhandling/InteractionValidatorPredicates.java

```diff
@@ -0,0 +1,37 @@
+package com.minecolonies.api.colony.interactionhandling;
+
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.util.Tuple;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.ITextComponent;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * Utility class to store all validator predicates for the chat handling.
+ */
+public final class InteractionValidatorPredicates
```

Comment:
This entire class needs to be done differently!
This implementation exposes the inner workings of the class instead of just providing an API surface to use.
Either turn this into an interface and stick an implementation in the API.

Or create methods around these maps, and return immutable ones plus add a register method to make it work.
Additionally this could potentially also be a simple IForgeRegistry for each with a callback.

TLDR: This needs to be different and and an actual registry.

Reply:
Ai no, creating a forge registry for this is an enormous overkill. I can conform to the access method part though.

Reply:
done